swap/src/lib.rs

```diff
@@ -1,7 +1,5 @@
 #![warn(
```

Comment:
Be sure you have the exact same compiler attributes in `main.rs`

Reply:
could remove the lib.rs now I think?

Reply:
After reviewing, I'd suggest to not worry about the lints and lib.rs within the scope of this PR.